pandas/io/html.py

```diff
@@ -831,9 +832,13 @@ def _parser_dispatch(flavor):
             raise ImportError(
                 "BeautifulSoup4 (bs4) not found, please install it")
         import bs4
-        if LooseVersion(bs4.__version__) <= LooseVersion('4.2.0'):
-            raise ValueError("A minimum version of BeautifulSoup 4.2.1 "
-                             "is required")
+        minimum_bs4_version = VERSIONS['bs4']
```

Comment:
Does `bs4 = import_optional_dependency('bs4')` work here? I don't recall why I didn't use it.

Reply:
So this actually done above:

```
def _importers():
    # import things we need
    # but make this done on a first use basis

    global _IMPORTS
    if _IMPORTS:
        return

    global _HAS_BS4, _HAS_LXML, _HAS_HTML5LIB
    bs4 = import_optional_dependency("bs4", raise_on_missing=False,
                                     on_version="ignore")
    _HAS_BS4 = bs4 is not None
```

But I think you're right; `bs4 = import_optional_dependency('bs4')` can be called here again.

Reply:
Ah so this wasn't changed because `import_optional_dependency` will raise a `ImportError` while this path raises a `ValueError` (there's a test for this).

I think it's worth changing to an `ImportError`. Thoughts @TomAugspurger?